.github/workflows/docs.yaml

```diff
@@ -0,0 +1,52 @@
+name: Docs Preview
+
+on:
+  pull_request:
+
+jobs:
+  preview_docs:
+    timeout-minutes: 30
+    name: Docs preview
+    if: "github.event_name == 'pull_request'"
+    runs-on: ubuntu-latest
+    env:
+      RUSTC_WRAPPER: "sccache"
+      SCCACHE_GHA_ENABLED: "on"
+    steps:
+    - uses: actions/checkout@v4
+    - uses: dtolnay/rust-toolchain@master
+      with:
+        toolchain: nightly-2024-05-02
```

Comment:
why nightly?

Reply:
Hm. This is copied from the `check_docs` job in `ci.yaml`. I don't know?

Reply:
try changing back to stable, i can't see any reason in the commit that switched to nightly and in general we only want to be on nightly until the specific issue is fixed in stable.  nightly get too outdated otherwise.

Reply:
Now we know why:
```
error[E0554]: `#![feature]` may not be used on the stable release channel
  --> iroh/src/lib.rs:90:21
   |
90 | #![cfg_attr(docsrs, feature(doc_cfg))]
   |                     ^^^^^^^^^^^^^^^^
```

Reply:
Great!  Can you make sure this is described in the commit message?  (the PR description becomes the commit message)

Reply:
Added something to the PR description :v: